WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt

```diff
@@ -165,15 +164,15 @@ class WPMainNavigationView @JvmOverloads constructor(
         }
 
         val fragment = navAdapter.getFragment(position)
+        val previousFragment = navAdapter.getFragment(prevPosition)
         if (fragment != null) {
-            fragmentManager
-                    .beginTransaction()
-                    .replace(R.id.fragment_container, fragment, getTagForPosition(position))
-                    // This is used because the main activity sometimes crashes because it's trying to switch fragments
-                    // after `onSaveInstanceState` was already called. This is the related issue
-                    // https://github.com/wordpress-mobile/WordPress-Android/issues/10852
-                    .commitAllowingStateLoss()
+            if (previousFragment != null) {
```

Comment:
AFAICT the changes won't change when the `updateCurrentPosition` gets invoked. So if it was invoked after `onSaveInstanceState` before these changes, it'll still be getting invoked after `onSaveInstanceState` now. I'm not a fan of using `commitAllowingStateLoss`, but I think not using it will re-introduce the crash. Wdyt?

Having said that it might be better to reintroduce the crash and fix it at its root rather than don't know about users who see empty fragments when they click on the bottom navigation. 🤷‍♂️

Reply:
In theory you're right and this still might happen. I agree I'd keep the crash and see if we can fix it properly when it happens. I think it was very rare even before.